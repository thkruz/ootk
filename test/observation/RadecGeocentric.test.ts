import { DEG2RAD, Degrees, EpochUTC, J2000, Kilometers, KilometersPerSecond, RadecGeocentric, Radians, RadiansPerSecond,
  Vector3D } from './../../src/main';
describe('RadecGeocentric', () => {
  let radec: RadecGeocentric;
  let exampleDate: Date;

  beforeEach(() => {
    exampleDate = new Date(1705109326817);
    radec = new RadecGeocentric(
      EpochUTC.fromDateTime(exampleDate),
      25 * DEG2RAD as Radians,
      0 * DEG2RAD as Radians,
    );
  });

  it('should be constructable', () => {
    expect(radec).toMatchSnapshot();
  });

  // fromDegrees
  it('should be constructable from degrees', () => {
    const radecDeg = RadecGeocentric.fromDegrees(
      EpochUTC.fromDateTime(exampleDate),
      25 as Degrees,
      0 as Degrees,
    );

    expect(radecDeg).toMatchSnapshot();
  });

  // fromStateVector
  it('should be constructable from a state vector', () => {
    const state = new J2000(
      EpochUTC.fromDateTime(exampleDate),
      new Vector3D<Kilometers>(7000 as Kilometers, 7000 as Kilometers, 8000 as Kilometers),
      new Vector3D<KilometersPerSecond>(0 as KilometersPerSecond, 0 as KilometersPerSecond, 0 as KilometersPerSecond),
    );
    const radecSv = RadecGeocentric.fromStateVector(state);

    expect(radecSv).toMatchSnapshot();
  });

  // rightAscensionDegrees
  it('should return the right ascension in degrees', () => {
    expect(radec.rightAscensionDegrees).toMatchSnapshot();
  });

  // declinationDegrees
  it('should return the declination in degrees', () => {
    expect(radec.declinationDegrees).toMatchSnapshot();
  });

  // rightAscensionRateDegrees
  it('should return the right ascension rate in degrees', () => {
    expect(radec.rightAscensionRateDegrees).toMatchSnapshot();
  });

  // declinationRateDegrees
  it('should return the declination rate in degrees', () => {
    expect(radec.declinationRateDegrees).toMatchSnapshot();
  });

  // position
  it('should return the position', () => {
    expect(radec.position()).toMatchSnapshot();
  });

  // velocity
  it('should return the velocity', () => {
    expect(() => radec.velocity()).toThrow();

    const radec2 = new RadecGeocentric(
      EpochUTC.fromDateTime(exampleDate),
      25 * DEG2RAD as Radians,
      1 * DEG2RAD as Radians,
      1000 as Kilometers,
      1.5 * DEG2RAD as RadiansPerSecond,
      2.5 * DEG2RAD as RadiansPerSecond,
    );

    expect(radec2.velocity()).toMatchSnapshot();
  });

  // angle
  it('should return the angle', () => {
    const radec2 = new RadecGeocentric(
      EpochUTC.fromDateTime(exampleDate),
      25 * DEG2RAD as Radians,
      1 * DEG2RAD as Radians,
    );

    expect(radec.angle(radec2)).toMatchSnapshot();
  });

  // angleDegrees
  it('should return the angle in degrees', () => {
    const radec2 = new RadecGeocentric(
      EpochUTC.fromDateTime(exampleDate),
      25 * DEG2RAD as Radians,
      1 * DEG2RAD as Radians,
    );

    expect(radec.angleDegrees(radec2)).toMatchSnapshot();
  });
});
