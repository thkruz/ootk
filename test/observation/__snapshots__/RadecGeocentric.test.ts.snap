// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RadecGeocentric should be constructable 1`] = `
RadecGeocentric {
  "declination": 0,
  "declinationRate": undefined,
  "epoch": EpochUTC {
    "posix": 1705109326.817,
  },
  "range": undefined,
  "rangeRate": undefined,
  "rightAscension": 0.4363323129985824,
  "rightAscensionRate": undefined,
}
`;

exports[`RadecGeocentric should be constructable from a state vector 1`] = `
RadecGeocentric {
  "declination": 0.679673818908244,
  "declinationRate": 0,
  "epoch": EpochUTC {
    "posix": 1705109326.817,
  },
  "range": 12727.922061357855,
  "rangeRate": 0,
  "rightAscension": 0.7853981633974483,
  "rightAscensionRate": -0,
}
`;

exports[`RadecGeocentric should be constructable from degrees 1`] = `
RadecGeocentric {
  "declination": 0,
  "declinationRate": null,
  "epoch": EpochUTC {
    "posix": 1705109326.817,
  },
  "range": undefined,
  "rangeRate": undefined,
  "rightAscension": 0.4363323129985824,
  "rightAscensionRate": null,
}
`;

exports[`RadecGeocentric should return the angle 1`] = `0.017453292519941554`;

exports[`RadecGeocentric should return the angle in degrees 1`] = `0.9999999999999002`;

exports[`RadecGeocentric should return the declination in degrees 1`] = `0`;

exports[`RadecGeocentric should return the declination rate in degrees 1`] = `null`;

exports[`RadecGeocentric should return the position 1`] = `
Vector3D {
  "x": 0.9063077870366499,
  "y": 0.42261826174069944,
  "z": 0,
}
`;

exports[`RadecGeocentric should return the right ascension in degrees 1`] = `25`;

exports[`RadecGeocentric should return the right ascension rate in degrees 1`] = `null`;

exports[`RadecGeocentric should return the velocity 1`] = `
Vector3D {
  "x": -11.752592909334684,
  "y": 23.401642759234754,
  "z": 43.62658574738897,
}
`;
